### PR TITLE
feat: Update Instructor dashboard Open Responses tab to be aware of ESG

### PR DESCRIPTION
**Pull request by leangseu-edx**
**TL;DR -** 
The Instructor dashboard needs to be aware of the ESG waffle flag to provide either the old ORA experience or the new ESG experience at a course by course basis.

JIRA: [AU-52](https://openedx.atlassian.net/browse/AU-52)

**What changed?**

- ORA js fragment now receive `ENHANCED_STAFF_GRADER` from js context

**Developer Checklist**

- [x] Reviewed the [release process](https://github.com/edx/edx-ora2/blob/master/.github/release_process.md)
- [x] Translations and JS/SASS compiled
- [x] Bumped version number in [setup.py](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/setup.py#L39) and [package.json](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/package.json#L3)

**Testing Instructions**

- add `console.log(this.data.ENHANCED_STAFF_GRADER)` to [this line](https://github.com/edx/edx-ora2/blob/bd7483697c42515e2af3f0d68eb09a056d2ff9b4/openassessment/xblock/static/js/src/lms/oa_course_items_listing.js#L235)
- Create a test course
- Confirm or toggle the waffle flag `openresponseassessment.enhanced_staff_grader` on for the course. Assuming this is the flag from [AU-50](https://openedx.atlassian.net/browse/AU-50)
- Navigate to the Instructor dashboard -> Open Response
- The console should log `true`
- Toggle the waffle flag `openresponseassessment.enhanced_staff_grader` off for the course
- Navigate to the Instructor dashboard -> Open Response
- The console should log `false`

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

File: openassessment/xblock/test/test_openassessment.py [hunk 1]
```diff
@@ -155,11 +155,13 @@ def test__create_ui_models__no_leaderboard_if_teams_enabled(self, xblock):
...
 
     @scenario('data/basic_scenario.xml')
-    def test_ora_blocks_listing_view(self, xblock):
+    @patch('openassessment.xblock.config_mixin.ConfigMixin.is_feature_enabled')
+    def test_ora_blocks_listing_view(self, xblock, mock_feature_enabled):
         """
         Test view for listing all courses OA blocks.
 
         """
+        mock_feature_enabled.return_value = False
```
[the remark this answers is not in the document]
Comment (leangseu-edx) [reply openassessment/xblock/test/test_openassessment.py]: The mock is require for rendering now. The use of `flag` require [this method(https://github.com/edx/edx-ora2/blob/008cfb6b65248bc19c44190e0ee54d48e54c69b7/openassessment/xblock/config_mixin.py#L44) but `openedx` is not define and I am not sure how to mock it either.

File: openassessment/xblock/test/test_openassessment.py [hunk 2]
```diff
@@ -156,6 +157,10 @@ def test__create_ui_models__no_leaderboard_if_teams_enabled(self, xblock):
         self.assertEqual(models[1], UI_MODELS["grade"])
 
     @scenario('data/basic_scenario.xml')
+    @patch(
```
Comment (jansenk) [inline openassessment/xblock/test/test_openassessment.py:160]: Is this necessary?

File: openassessment/xblock/test/test_openassessment.py [hunk 3]
```diff
@@ -202,6 +207,33 @@ def test_ora_blocks_listing_view(self, xblock):
         items = json.loads(scripts[0].text)
         self.assertEqual(items, defined_ora_items)
 
+    @scenario('data/basic_scenario.xml')
+    @ddt.data(
+        *list(itertools.product([True, False]))
+    )
+    @ddt.unpack
```
Comment (jansenk) [inline openassessment/xblock/test/test_openassessment.py:214]: I might be missing something, but this looks like it's just a complicated way of doing
```suggestion
    @ddt.data(False, True)
```

**Review by jansenk — CHANGES_REQUESTED** [review]
Some test changes but this looks like it does what the ticket asks

File: openassessment/xblock/test/test_openassessment.py (hunk 2 above)
Comment (leangseu-edx) [reply openassessment/xblock/test/test_openassessment.py:160]: `openedx` code suppose to get run and it is not define in a normal environment. There isn't safety around it either. The test would through error if I do not mock it.

Comment (jansenk) [reply openassessment/xblock/test/test_openassessment.py:159]: I see I already asked this question several days ago xD

**Comment by jansenk** [comment]
Resolve the conflicts and I'll give you a thumb

**Comment by leangseu-edx** [comment]
> Resolve the conflicts and I'll give you a thumb

Done 👍

**Review by jansenk — APPROVED** [review]
